test/AddressSpaceCast/to_global.cl

```diff
@@ -0,0 +1,18 @@
+// RUN: clspv --cl-std=CLC++ --inline-entry-points %s -o %t.spv --show-producer-ir &> %t.ll
+// RUN: spirv-val --target-env spv1.0 %t.spv
+// RUN: FileCheck %s < %t.ll
+
+// CHECK-NOT: addrspacecast
+
+float loop(const float *data, unsigned num) {
+    float res = 0;
+    for (unsigned j = 0; j < num; ++j) {
+        res += to_local(data)[j];
```

Comment:
This seems wrong? The test name is to_global and the pointer that is passed in here is a global pointer. Are we actually generating correct code?

Reply:
Yes, this is an error.
The code generated was "correct", it was doing the same thing as the `incompatible_cast.ll` test.
When to call `to_<addrspace>` with an incompatible pointer, we just need to return `null`.